core/sql/optimizer/RelExpr.cpp

```diff
@@ -6630,6 +6630,23 @@ const NAString Intersect::getText() const
 }
 
 // -----------------------------------------------------------------------
+// // member functions for class Except
+// // -----------------------------------------------------------------------
+Except::Except(RelExpr *leftChild,
+             RelExpr *rightChild)
+: RelExpr(REL_EXCEPT, leftChild, rightChild)
+{ setNonCacheable(); }
```

Comment:
I don't know either, but SET operations like UNION is also set as non cacheable. So I feel we should keep it now. I will try to study more on this later.

showshape works:
>>showshape select eno,ename,dno from t021 except select * from t022;
I
control query shape hybrid_hash_join(scan(path 'TRAFODION.SEABASE.T021',
forward, blocks_per_access 1 , mdam off),
scan(path 'TRAFODION.SEABASE.T022', forward, blocks_per_access 1
, mdam off));

--- SQL operation complete.
>>